### DISCUSSION TITLE
rust-course/错误处理中的组合器部分 应该更通俗一些

### DISCUSSION THREAD

**Discussion by LyntNy4n** [sunface/rust-course#596] [内容建议]
目前是把or_else() 和 and_then()等函数的使用情况都列举出来 但是不直观也让人看不下去
应该是介绍这些函数的应用场景 比如and_then()是取Ok或Option里的内容并执行闭包里的代码,如果是Err或None就不管 这个函数就是关注有结果的东西而不是错误或者空值
或者可以写出等效的match表达式 让别人知道这个函数简化了哪些工作

**Comment by sunface** [comment]
非常有道理。。。目前的内容确实更像是: 有这个东西，而不是有这个东西大家可以用来做什么。

**Comment by sunface** [comment]
我转到 issue 了
